src/h5web/visualizations/shared/AxisSystem.tsx

```diff
@@ -45,27 +55,17 @@ function AxisSystem(props: Props): JSX.Element {
     ]);
   });
 
-  const sharedAxisProps = {
-    tickStroke: 'grey',
-    hideAxisLine: true,
-    tickFormat: format('0'),
-    tickClassName: styles.tick,
-    tickComponent: ({ formattedValue, ...tickProps }: TickRendererProps) => (
-      <text {...tickProps}>{formattedValue}</text>
-    ),
-  };
-
-  // Restrain axis scales to visible domains
-  const xAxisScale = abscissa.scaleFn();
-  xAxisScale.domain(visibleDomains[0]);
-  xAxisScale.range([0, width]);
+  // Restrain ticks scales to visible domains
+  const xTicksScale = abscissa.scaleFn();
+  xTicksScale.domain(visibleDomains[0]);
+  xTicksScale.range([0, width]);
 
-  const yAxisScale = ordinate.scaleFn();
-  yAxisScale.domain(visibleDomains[1]);
-  yAxisScale.range([height, 0]);
+  const yTicksScale = ordinate.scaleFn();
+  yTicksScale.domain(visibleDomains[1]);
+  yTicksScale.range([height, 0]);
 
-  const abscissaTicksNumber = adaptedNumTicks(width);
-  const ordinateTicksNumber = adaptedNumTicks(height);
+  const xTicksProp = getTicksProp(abscissaConfig, visibleDomains[0], width);
+  const yTicksProp = getTicksProp(ordinateConfig, visibleDomains[1], height);
```

Comment:
`getTicksProp` is just a helper function that returns an object with either `numTicks` (data axis) or `tickValues` (index axis). The object is then destructured on both the axis and grid components.